pennylane/hf/integrals.py

```diff
@@ -104,3 +103,179 @@ def contracted_norm(l, alpha, a):
+    """
+    basis_params = []
+    c = 0
+    for p in params:
+        if p.requires_grad:
+            basis_params.append(args[c])
+            c += 1
+        else:
+            basis_params.append(p)
+    return tuple(basis_params)
+
+
+def expansion(la, lb, ra, rb, alpha, beta, t):
+    r"""Compute Hermite Gaussian expansion coefficients recursively for two Gaussian functions.
+
+    An overlap distribution, which defines the product of two Gaussians, can be written as a Hermite
+    expansion as [`Helgaker (1995) p798 <https://www.worldscientific.com/doi/abs/10.1142/9789812832115_0001>`_]
+
+    .. math::
+
+        \Omega_{ij} = \sum_{t=0}^{i+j} E_t^{ij} \Lambda_t,
+
+    where :math:`\Lambda` is a Hermite polynomial of degree :math:`t`, :math:`E` denotes the expansion
+    coefficients, :math:`\Omega_{ij} = G_i G_j`, and :math:`G` is a Gaussian function. The overalp
+    integral between two Gaussian functions can be simply computed by integrating over the overlap
+    distribution which requires obtaining the expansion coefficients. This can be done recursively
+    as [`Helgaker (1995) p799 <https://www.worldscientific.com/doi/abs/10.1142/9789812832115_0001>`_]
+
+    .. math::
+
+        E_t^{i+1,j} = \frac{1}{2p} E_{t-1}^{ij} - \frac{qr}{\alpha} E_{t}^{ij} + (t+1) E_{t+1}^{ij},
+
+    and
+
+    .. math::
+
+        E_t^{i,j+1} = \frac{1}{2p} E_{t-1}^{ij} + \frac{qr}{\beta} E_{t}^{ij} + (t+1) E_{t+1}^{ij},
+
+    where :math:`p = \alpha + \beta` and :math:`q = \alpha \beta / (\alpha + \beta)` are computed
+    from the Gaussian exponents :math:`\alpha, \beta` and the position :math:`r` is computed as
+    :math:`r = r_\alpha - r_\beta`. The starting coefficient is
+
+    .. math::
+
+        E_0^{00} = e^{-qr^2},
+
+    and :math:`E_t^{ij} = 0` is :math:`t < 0` or :math:`t > (i+j)`.
+
+    Args:
+        la (integer): angular momentum component for the first Gaussian function
+        lb (integer): angular momentum component for the second Gaussian function
+        ra (float): position component of the the first Gaussian function
+        rb (float): position component of the the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+        t (integer): number of nodes in the Hermite Gaussian
+
+    Returns:
+        array[float]: expansion coefficients for each Gaussian combination
+
+    **Example**
+
+    >>> la, lb = 0, 0
+    >>> ra, rb = 0.0, 0.0
+    >>> alpha = np.array([3.42525091])
+    >>> beta =  np.array([3.42525091])
+    >>> t = 0
+    >>> c = expansion(la, lb, ra, rb, alpha, beta, t)
+    >>> c
+    array([1.])
+    """
+    p = alpha + beta
+    q = alpha * beta / p
+    r = ra - rb
+
+    if la == lb == t == 0:
+        return anp.exp(-q * r ** 2)
+
+    if t < 0 or t > (la + lb):
+        return 0.0
+
+    elif lb == 0:
+        return (
```

Comment:
@soranjh it looks like linting has not been run on this file yet, since this `elif` should be replaced with `if` statements (and similar for the `else` below)?

Reply:
Fixed.